### PR TITLE
Subarray sum

### LINKED ISSUE #16: Challenge: Subarray sum
Write a python function to sum the elements of a sub-array from a given array.

### PR DESCRIPTION
**Pull request by Fahim-Ahmad**
- [x] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [x] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [ ] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [x] The markdown looks correct when you preview the file
- [x] All links and images work
- [x] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [ ] There is a module header
- [x] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [x] Parameter description
- [x] Return value description
- [x] Include any assumptions
- [x] Include 3 or more (passing!) doctests
- [x] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [ ] There are type annotations
- [ ] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Each unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [x] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/brain_teasers/python/subarray_sum/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Sum of subarray
```
Comment (mohammedalasly) [inline src/brain_teasers/python/subarray_sum/README.md:1]: this README file is good, just needs to be more readable by adding Python after the backticks

```python
_your code_
```
you can also add the command line that is used to test your implementation and a conclusion at the end

**Review by mohammedalasly — CHANGES_REQUESTED** [review]

File: src/brain_teasers/python/subarray_sum/src/subarray_sum.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+def subarray_sum(arr, start_index, end_index):
```
Comment (mohammedalasly) [inline src/brain_teasers/python/subarray_sum/src/subarray_sum.py:6]: This code passed the test successfully in the local environment, just needs a Module header name to be added

**Review by mohammedalasly — CHANGES_REQUESTED** [review]

File: src/brain_teasers/python/subarray_sum/README.md (hunk 1 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/subarray_sum/README.md:1]: Thanks, I made below changes in three different commit so it will be easy to check:

- added the word "python" after the backticks
- added the instruction for testing the code implementation
- added a summary at the end

File: src/brain_teasers/python/subarray_sum/src/subarray_sum.py (hunk 2 above)
Comment (Fahim-Ahmad) [reply src/brain_teasers/python/subarray_sum/src/subarray_sum.py:6]: Thanks for noticing it. The module header is added now.

File: src/brain_teasers/python/subarray_sum/README.md (hunk 1 above)
Comment (mohammedalasly) [reply src/brain_teasers/python/subarray_sum/README.md:1]: Nice work!!
